lib/class-wp-patterns-registry.php

```diff
@@ -82,7 +82,7 @@ public function get_registered( $pattern_name ) {
 	 *               and per style.
 	 */
 	public function get_all_registered() {
-		return array_values( $this->registered_patterns );
+		return $this->registered_patterns;
```

Comment:
I think the return value should still be a regular array here, what if we change the registration instead to add the "name" property to the pattern when registered? It's more inline with what we do for blocks I believe.

Reply:
This matches exactly how the `WP_Block_Styles_Registry` functions.

https://github.com/WordPress/gutenberg/blob/eb4d81fdaac120b24e53d7968ec340e6cb63a95f/lib/class-wp-block-styles-registry.php#L104-L113

Though, open to what you think will scale best. Not sold on any particular ways of doing this.

Reply:
I'm  back at this again, I believe  it's better to map block registration instead and use a "name" property for the slug and return an array (and not associative one) here.

Reply:
So just so I understand you correctly, you're suggesting that `register_pattern` should take only an array containing `name`, `title`, and `content`?

Reply:
no, more  leasing the function as is but its implementation should embed the "name" inside the "settings" object. like we do here https://github.com/WordPress/gutenberg/blob/master/packages/blocks/src/api/registration.js#L144

Reply:
Got it. Sounds reasonable.